components/wifi/esp8266-driver/mbed_lib.json

```diff
@@ -21,6 +21,15 @@
             "help": "RESET pin for the modem, defaults to Not Connected",
             "value": null
         },
+        "rst-pin-polarity": {
+            "help": "Polarity of RESET pin for the modem. [0/1]",
+            "options": [0, 1],
+            "value": 0
+        },
+        "rst-assert-time-us": {
+            "help": "Assert time of reset for the modem in us.",
+            "value": 2000
```

Comment:
Why is this in microseconds, when it's really interpreted as "2 ticks", meaning "at least 1000us"? If this is going to be microseconds, then it should be set to 200, and the code should be rounding that up to 2 ticks: `(us + 999) / 1000 + 1` = `(us + 1999) / 1000`.

But again, why is it configurable? Why would this differ between hardware? If the 3000 number for your target is correct, may as well always use it? (How many real microseconds is that supposed to represent? 2000?)

Reply:
Original code is 2 ms:
```
ThisThread::sleep_for(2);
```
So I change to:
```
ThisThread::sleep_for(MBED_CONF_ESP8266_RST_ASSERT_TIME_US / 1000);
```
where `MBED_CONF_ESP8266_RST_ASSERT_TIME_US` defaults to 2000 us.
But it seems the code doesn't follow the doc.

Configuring `MBED_CONF_ESP8266_RST_ASSERT_TIME_US` to 3000 for M2351 is for power on/off control. Its discussion starts from #11299.

Reply:
Right, but the point is that it isn't trying to delay for 2000us. It's trying to delay for at least 200us. If you wanted a 2000us delay, it would have to wait for 3 ticks.

Using microseconds as the unit, strongly suggests you're fishing for some sort of microsecond precision, so I would immediately question the code as written - pointing out that `sleep_for(time_us / 1000)` may wait for only 1000us, not the 2000us specified.

If this isn't really the ESP8266's reset pin, but a module power pin, then it would be better to add a new setting for the new pin, rather than hijack the code thinking that it's driving reset. That would be in line with what we're doing for all the cellular modems - one "soft" power line which is the device's own "button", and one "hard" which is a power control.